src/serdes/binary-parser.ts

```diff
@@ -107,10 +107,26 @@ class BinaryParser {
    * @return Field ordinal
    */
   readFieldOrdinal(): number {
-    const tagByte = this.readUInt8();
-    const type = (tagByte & 0xf0) >>> 4 || this.readUInt8();
-    const nth = tagByte & 0x0f || this.readUInt8();
-    return (type << 16) | nth;
+    let type = this.readUInt8();
+    let field = type & 15;
+    type >>= 4;
+
+
+    if(type === 0) {
+      type = this.readUInt8();
+      if(type === 0 || type < 16) {
```

Comment:
```suggestion
      if (type === 0 || type < 16) {
```
Later on, we could set up Prettier or some linter to make sure this kind of spacing is consistent

Reply:
Just forgot to run the linter